File: design-patterns/go/composite/player_group/football_player.go
// football_player.go

package main

import "fmt"

type FootballPlayer struct {
	name string
	age  int
	goal int
}

func NewFootballPlayer(name string, age int, goal int) (footballPlayer *FootballPlayer) {
	footballPlayer = &FootballPlayer{}
	footballPlayer.name = name
	footballPlayer.age = age
	footballPlayer.goal = goal
	return
}

func (footballPlayer *FootballPlayer) PrintDetails() {
	fmt.Println("Game: Football")
	fmt.Printf("Name: %v\n", footballPlayer.name)
	fmt.Printf("Age: %v\n", footballPlayer.age)
	fmt.Printf("Goals: %v\n", footballPlayer.goal)
}

File: design-patterns/php/composite/playergroup/FootballPlayer.php
<?php
// FootballPlayer.php

namespace BigBoxCode\DesignPattern\Composite\PlayerGroup;


class FootballPlayer implements Player {
    public function __construct(
        private string $name,
        private int $age,
        private int $goal
    ) {

    }

    public function printDetails(): void {
        echo "\nGame: Football";
        echo "\nName: " . $this->name;
        echo "\nAge: " . $this->age;
        echo "\nGoals: " . $this->goal;
    }
}